astropy/io/fits/connect.py

```diff
@@ -302,3 +306,202 @@ def write_table_fits(input, output, overwrite=False):
 io_registry.register_reader('fits', Table, read_table_fits)
 io_registry.register_writer('fits', Table, write_table_fits)
 io_registry.register_identifier('fits', Table, is_fits)
+
+
+def read_data_fits(filename, ext_data=0, ext_meta=0, ext_mask='mask',
```

Comment:
Is it specific to 2D data? I checked locally and scalar, 1D, 2D and 3D work if I just write them to a file and read it again. Unfortunatly I only have 2D science FITS files so I cannot comment if something fails if given a 1D or 3D file.